content/docs/command-reference/data/index.md

```diff
@@ -0,0 +1,24 @@
+# data
+
+Contains a command that shows changes in data tracked by dvc:
+[status](/doc/command-reference/data/status).
+
+## Synopsis
+
+```usage
+usage: dvc data [-h] [-q | -v] {status} ...
+
+positional arguments:
+  COMMAND
+    status         Show changes between the last git commit, the dvcfiles and the workspace.
+```
+
+## Description
+
+DVC discovers data tracked by DVC using the file path and the file hash
+specified in the `.dvc` and `dvc.lock` files, and builds an index out of it.
+
+This is used by DVC, for example, to show `dvc data status`, by comparing
+different versions of the index. DVC uses <abbr>cache</abbr> to compare between
+the specified hashes, the workspace and the actual file present in the cache to
+see if they have changed.
```

Comment:
> not sure what to add in data/index.md, so I started to introduce a few concepts around index/dvc-data, which is going to be helpful in the data/status.md ... mention the term index

Good use for the index for now. We'll probably want to move this to some user guide at some point. Probably rel. https://github.com/iterative/dvc.org/issues/2856

Reply:
> and builds an index out of it

What kind of index? How is it built? (Index contents) Why is it needed if the info is already in metafiles? Is it the same as [.dvc/tmp/index](https://dvc.org/doc/user-guide/project-structure/internal-files)? Is it committed? Is it cached?

Possible edit, please check for accuracy:

```suggestion
## Description

DVC tracks data files using the `path` and hash value (`md5`, `etag`, or
`checksum`) saved in `.dvc` and `dvc.lock` metafiles. An index is
kept with this information to optimize data management operations.

This is used, for example, to show the `dvc data status` -- by comparing
different versions of the index (the last one committed to Git and the one
present in the worksapce) against <abbr>cached</abbr> data hashes.
```

Reply:
At the moment, it's a concept that does not exist in reality (although we feel the need for it and there will be one soon). 

It's only introduced so that we don't repeat "comparing the file hashes from dvc.lock and .dvc files" over and over, and that comparison happens between head and workspace, and the workspace and the files actually in the workspace.

Reply:
I also don't understand what kind of index does it build. Does it save it? Also feels like an implementation detail? Do we expect ppl to do anything with this index?

Reply:
should we just drop the index page for now and to simplify this?

Reply:
Here's the index: https://github.com/iterative/dvc-data/blob/main/src/dvc_data/index.py. It's an in-memory trie. We may serialize/deserialize it in the future.

We are migrating to it, so we are currently not using it right now. The docs for this is simple enough, and does not go too much into details that I think we can leave it as-is for now. 

> should we just drop the index page for now and to simplify this?

Can we do that? Is `index.md` not required?

Reply:
> It's only introduced so that we don't repeat "comparing the file hashes from dvc.lock and .dvc files" over and over, and that comparison happens between head and workspace, and the workspace and the files actually in the workspace.

It seems like `index` was removed except for this one instance. Can we drop the term here and otherwise leave as is?

```suggestion
## Description

DVC discovers data tracked by DVC using the file path and the file hash
specified in the `.dvc` and `dvc.lock` files.

This is used by DVC, for example, to show `dvc data status`, by comparing
between the specified hashes, the workspace and the actual file present in the cache to
see if they have changed.
```

Or combining with suggestions from @jorgeorpinel:

```suggestion
## Description

DVC tracks data files using the `path` and hash value (`md5`, `etag`, or
`checksum`) saved in `.dvc` and `dvc.lock` metafiles.

This is used, for example, to show the `dvc data status` -- by comparing
between the last version committed to Git and the one
present in the workspace against <abbr>cached</abbr> data hashes.
```

Reply:
Or drop `Description` completely?

Reply:
yes, index.md is optional

Reply:
again, not sure this page is valuable tbh ... at least for now